$NetBSD: patch-src_CMakeLists.txt,v 1.1 2023/10/21 11:20:23 nros Exp $

Not all unix ld commands support --version_script ,
check if it works before using it

--- src/CMakeLists.txt.orig	2023-10-21 11:00:42.582420428 +0000
+++ src/CMakeLists.txt
@@ -78,9 +78,10 @@ set_property(TARGET charls PROPERTY PUBL
 
 set_target_properties(charls PROPERTIES
   LINK_DEPENDS ${CMAKE_CURRENT_LIST_DIR}/charls.version)
-if(UNIX AND NOT APPLE)
-  set_property(TARGET charls APPEND_STRING PROPERTY
-    LINK_FLAGS " -Wl,--version-script=${CMAKE_CURRENT_LIST_DIR}/charls.version")
+include(CheckLinkerFlag)
+check_linker_flag(CXX "LINKER:--version-script=${CMAKE_CURRENT_LIST_DIR}/charls.version" versionscript_flag)
+if(versionscript_flag)
+  target_link_options(charls PRIVATE "LINKER:--version-script=${CMAKE_CURRENT_LIST_DIR}/charls.version")
 endif()
 
 target_sources(charls
